Integrate Azure Defender with Cluster API as an Azure Arc Connected Cluster using Kubernetes extensions (#534)
microsoft/azure_arc: docs/azure_arc_jumpstart/azure_arc_k8s/day2/cluster_api/cluster_api_defender_extension/_index.md

    ```shell
    az login
    az ad sp create-for-rbac -n "<Unique SP Name>" --role contributor
    ```

    For example:

    ```shell
    az ad sp create-for-rbac -n "http://AzureArcK8s" --role contributor
    ```

    Output should look like this:

    ```json
    {
    "appId": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "displayName": "AzureArcK8s",
    "name": "http://AzureArcK8s",
    "password": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "tenant": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX"
    }
    ```

    > **Note: The Jumpstart scenarios are designed with as much ease of use in-mind and adhering to security-related best practices whenever possible. It is optional but highly recommended to scope the service principal to a specific [Azure subscription and resource group](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest) as well considering using a [less privileged service principal account](https://docs.microsoft.com/en-us/azure/role-based-access-control/best-practices)**

## Automation Flow

For you to get familiar with the automation and deployment flow, below is an explanation.

> * User has deployed Cluster API.

Reviewer: ```suggestion
* User has deployed Kubernetes using Cluster API and has it as Azure Arc enabled Kubernetes cluster.
```

> * User is editing the environment variables on the Shell script file (1-time edit). These variables are being used throughout the extension deployment. User is running the shell script. The script will use the extension management feature of Azure Arc to theploy the Azure Defender extension on the Arc connected cluster.

Reviewer: ```suggestion
* User is editing the environment variables on the Shell script file (1-time edit) which then be used throughout the extension deployment. 

* User is running the shell script. The script will use the extension management feature of Azure Arc to deploy the Azure Defender extension on the Azure Arc connected cluster.
```

* User is veryfing the cluster is shown in Azure Defender and that the extension is deployed.

> * User is simulating a security alert on the Cluster API, Azure Defender detects this activity and triggers a security alert.

Reviewer: ```suggestion
* User is simulating a security alert on the Kubernetes cluster that will cause Azure Defender to detect this activity and trigger a security alert.
```

## Create Azure Defender extensions instance

To create a new extension Instance, we will use the _k8s-extension create_ command while passing in values for the mandatory parameters. This scenario provides you with the automation to deploy the Azure Defender extension on your Azure Arc enabled Kubernetes cluster.

* Before integrating the cluster with Azure Defender, click on the "Security (preview)" blade for the connected Arc cluster to show how the cluster is not currently being assessed by Azure Defender.

>     ![Screenshot showing Azure Portal with Azure Arc enabled Kubernetes resource](./01.png)

Reviewer: Mask subscription

Author: screenshot added

Author: fixed

    ![Screenshot showing Azure Portal with Azure Arc enabled Kubernetes resource extensions](./02.png)

> * Edit the environment variables in the script to match your environment parameters. After that run it using the ```. ./azure_defender_k8s_extension.sh``` command.

Reviewer: ```suggestion
* Edit the environment variables in the script to match your environment parameters followed by running the ```. ./azure_defender_k8s_extension.sh``` command.
```

    > **Note: The extra dot is due to the shell script has an *export* function and needs to have the vars exported in the same shell session as the rest of the commands.**

> 

Reviewer: Might be good to provide an example

Author: added screenshot

    ![Screenshot showing Cluster API in Azure Portal](./03.png)

    ![Screenshot showing Azure Arc connected cluster](./04.png)

>     The script will:
> 
>   * Login to your Azure subscription using the SPN credentials
>   * Add or Update your local _connectedk8s_ and _k8s-extension_ Azure CLI extensions
>   * Create defender k8s extension instance

Reviewer: ```suggestion
    The script will:

  * Login to your Azure subscription using the SPN credentials
  * Add or Update your local _connectedk8s_ and _k8s-extension_ Azure CLI extensions
  * Create Azure Defender k8s extension instance
```

> * You can see that Azure Defender is enabled once you visit the security tab section of the Azure Arc enabled Kubernetes cluster resource in Azure.

Reviewer: Screenshot?

Author: added the screenshot

* Also verify under the Extensions section of the Azure Arc enabled Kubernetes cluster that the Azure Defender extension is correctly installed.

![Screenshot extension deployment](./05.png)

## Simulate a security alert

> To verify that Azure Defender is working properly and alerting on security threats, you can run this command to simulate an alert:

Reviewer: ```suggestion
To verify that Azure Defender is working properly and alerting on security threats, run the below command to simulate an alert:
```

    ![Screenshot showing Azure Portal with Azure Arc enabled Kubernetes resource extensions](./02.png)

* Edit the environment variables in the script to match your environment parameters followed by running the ```. ./azure_defender_k8s_extension.sh``` command.

    ![Screenshot parameter examples](./03.png)

    > **Note: The extra dot is due to the shell script has an *export* function and needs to have the vars exported in the same shell session as the rest of the commands.**

    ![Screenshot showing Cluster API in Azure Portal](./04.png)

    ![Screenshot showing Azure Arc connected cluster](./05.png)

    The script will:

  * Login to your Azure subscription using the SPN credentials
  * Add or Update your local _connectedk8s_ and _k8s-extension_ Azure CLI extensions
  * Create Azure Defender k8s extension instance

* You can see that Azure Defender is enabled once you visit the security tab section of the Azure Arc enabled Kubernetes cluster resource in Azure.

* Also verify under the Extensions section of the Azure Arc enabled Kubernetes cluster that the Azure Defender extension is correctly installed.

![Screenshot extension deployment](./06.png)

> 

Reviewer: I think we should show what is deployed on the cluster

Author: added command and screenshot

## Simulate a security alert

To verify that Azure Defender is working properly and alerting on security threats, run the below command to simulate an alert:

```bash
kubectl get pods --namespace=asc-alerttest-662jfi039n --kubeconfig <cluster-name>.kubeconfig
```

Within 30 minutes Azure Defender will detect this event and trigger a security alert that you wil see in the Azure Portal under Azure Security Center's security alerts and also on the security tab of your Azure Arc enabled cluster.

![Screenshot security alert in Azure Security Center](./07.png)

![Screenshot security alert in Azure Security Center](./08.png)

![Screenshot security alert in Azure Security Center](./09.png)

### Delete extension instance

> The following command only deletes the extension instance, but doesn't delete the Log Analytics workspace or disables Azure Defender

Reviewer: ```suggestion
The following command only deletes the extension instance, but doesn't delete the Log Analytics workspace or disables Azure Defender.
```

    ![Screenshot showing Azure Portal with Azure Arc enabled Kubernetes resource extensions](./02.png)

* Edit the environment variables in the script to match your environment parameters followed by running the ```. ./azure_defender_k8s_extension.sh``` command.

    ![Screenshot parameter examples](./03.png)

    > **Note: The extra dot is due to the shell script has an *export* function and needs to have the vars exported in the same shell session as the rest of the commands.**

   The script will:

  * Login to your Azure subscription using the SPN credentials
  * Add or Update your local _connectedk8s_ and _k8s-extension_ Azure CLI extensions
  * Create Azure Defender k8s extension instance

* You can see that Azure Defender is enabled once you visit the security tab section of the Azure Arc enabled Kubernetes cluster resource in Azure.

![Screenshot extension deployment security tab](./04.png)

* Also verify under the Extensions section of the Azure Arc enabled Kubernetes cluster that the Azure Defender extension is correctly installed.

![Screenshot extension deployment](./05.png)

* You can also verify the deployment by running the command below:

```bash

> kubectl get pods -n azuredefender --kubeconfig $CAPI_WORKLOAD_CLUSTER_NAME.kubeconfig

Reviewer: ```suggestion
kubectl get pods -n azuredefender --kubeconfig $CAPI_WORKLOAD_CLUSTER_NAME.kubeconfig
```

Outside the script, the environment var will not work since it's in a different shell environment.

Author: changed to <cluster-name>.kubeconfig
